persist-cli/src/main/java/io/ballerina/persist/nodegenerator/BalSyntaxTreeGenerator.java

```diff
@@ -1111,7 +1165,7 @@ public static void generateRelations(ArrayList<Entity> entityArray) {
                                     relation.getReferences().add(modifiedKey);
                                 }
                             } else if (relation.getKeyColumns().isEmpty()) {
-                                for (String key : childEntity.getKeys()) {
```

Comment:
Why are we changing this to references?

Reply:
Here, KeyColumns is empty, but the reference contains a value. So we have to get the key from the reference.

Reply:
Here, KeyColumns is empty, but the reference contains a value. So we have to get the key from the reference.